lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,304 @@
+              set to C(false) used on personally controlled sites using self-signed
+              certificates.
+    useSharedLogonAuthentication:
+        default: false
+        description:
+            - Whether or not Shared Logon Authentication will be used.
+    useRadiusAuthentication:
+        default: false
+        description:
+            - Whether or not users will be authenticated via a RADIUS server. Valid values are true/false
+    cyberarkSession:
+        description:
+            - Dictionary set by a CyberArk authentication containing the different values to perform actions on a logged-on CyberArk session
+'''
+
+EXAMPLES = '''
+- name: Logon to CyberArk Vault using PAS Web Services SDK - UseSharedLogonAuthentication
+  cyberark_authentication:
+    apiBaseUrl: "{{ WebServicesBaseURL }}"
+    useSharedLogonAuthentication: true
+
+- name: Logon to CyberArk Vault using PAS Web Services SDK - Not UseSharedLogonAuthentication
+  cyberark_authentication:
+    apiBaseUrl: "{{ WebServicesBaseURL }}"
+    username: "{{ PasswordObject.password }}"
+    password: "{{ PasswordObject.passprops.username }}"
+    useSharedLogonAuthentication: false
+
+- name: Logoff from CyberArk Vault
+  cyberark_authentication:
+    state: absent
+    cyberarkSession: "{{ cyberarkSession }}"
+'''
+
+RETURN = '''
+cyberarkSession:
+    description: Authentication facts.
+    returned: success
+    type: dictionary
+    contains:
+        token:
+            description: Session Token returned by the logon operation
+            returned: success
+            type: string
+            sample: "AAEAAAD/////AQAAAAA......NiNWMtNDhhNy00ZDc5LWE2MTQtMmRlMTBjMWI1ZWQ2BgYAAAABNAs="
+        apiBaseUrl:
+            description: A string containing the base URL of the server hosting CyberArk's Privileged Account Security Web Services SDK.
+            returned: success
+            type: string
+            sample: "https://<IIS_Server_PVWA>"
+        validate_certs:
+            description: Whether SSL certificates will be validated.
+            returned: success
+            type: bool
+            sample: true
+        useSharedLogonAuthentication:
+            description: Whether or not Shared Logon Authentication will be used.
+            returned: success
+            type: bool
+            sample: true
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+import urllib
+import httplib
+import traceback
+import sys
+import json
+
+
+def processAuthentication(module):
+
+    apiBaseUrl = module.params["apiBaseUrl"]
+    validate_certs = module.params["validate_certs"]
+    username = module.params["username"]
+    password = module.params["password"]
+    newPassword = module.params["newPassword"]
+    useSharedLogonAuthentication = module.params[
+        "useSharedLogonAuthentication"]
+    useRadiusAuthentication = module.params["useRadiusAuthentication"]
```

Comment:
:-) ... I am all for either one... initially was camelCase, and then started diverging when I had to change it for validateCerts (validate_certs). 

Changing everything to be consistent.